src/components/tl/ucp/allreduce/allreduce_sliding_window.c

```diff
@@ -11,80 +11,505 @@
+        pipe->put_requests[i] = NULL;
+    }
 }
 
 ucc_status_t
-ucc_tl_ucp_allreduce_sliding_window_start(ucc_coll_task_t __attribute__((unused)) *coll_task) //NOLINT
+ucc_tl_ucp_allreduce_sliding_window_start(ucc_coll_task_t *coll_task)
 {
-    return UCC_OK;
+    ucc_base_coll_args_t *coll_args   = &coll_task->bargs;
+    ucc_schedule_t       *schedule    = ucc_derived_of(coll_task,
+                                                       ucc_schedule_t);
+    ucc_base_team_t      *base_team   = schedule->super.team;
+    ucc_tl_ucp_team_t    *team        = ucc_derived_of(base_team,
+                                                       ucc_tl_ucp_team_t);
+    ucc_tl_ucp_context_t *tl_ctx      = UCC_TL_UCP_TEAM_CTX(team);
+    ucc_rank_t            rank        = UCC_TL_TEAM_RANK(team);
+    uint32_t              count_total = coll_task->bargs.args.dst.info.count;
+    ucc_rank_t            size        = coll_task->team->params.size;
+    ucc_datatype_t        dtype       = coll_args->args.dst.info.datatype;
+    size_t                dt_size     = ucc_dt_size(dtype);
+    int                   inplace     = UCC_IS_INPLACE(coll_args->args);
+    ucc_status_t          status      = UCC_OK;
+    int                   put_window_size = UCC_TL_UCP_TEAM_LIB(team)
+                                 ->cfg.allreduce_sliding_window_put_window_size;
+    ucc_tl_ucp_allreduce_sw_global_work_buf_info_t *gwbi_p =
+        coll_args->args.global_work_buffer;
+    ucc_tl_ucp_task_t *rdma_task = ucc_derived_of(schedule->tasks[0],
+                                                  ucc_tl_ucp_task_t);
+    ucc_tl_ucp_allreduce_sw_pipeline_t       *pipe;
+    ucc_tl_ucp_allreduce_sw_host_allgather_t *allgather_data;
+
+    pipe           = rdma_task->allreduce_sliding_window.pipe;
+    allgather_data = rdma_task->allreduce_sliding_window.allgather_data;
+
+    // Register the src buf
+    if (!inplace) {
+        status = ucc_tl_ucp_allreduce_sliding_window_register(
+            tl_ctx->worker.ucp_context, team,
+            rdma_task->allreduce_sliding_window.bufs->src_ebuf,
+            gwbi_p->packed_src_memh);
+        if (status != UCC_OK) {
+            tl_error(UCC_TASK_LIB(rdma_task), "failed to register src memh: %s",
+                        ucc_status_string(status));
+            goto out;
+        }
+        memcpy(allgather_data->packed_src_key,
+               rdma_task->allreduce_sliding_window.bufs->src_ebuf->packed_key,
+               rdma_task->allreduce_sliding_window.bufs->src_ebuf->packed_key_len);
+    }
+
+    // Register the dst buf
+    status = ucc_tl_ucp_allreduce_sliding_window_register(
+        tl_ctx->worker.ucp_context, team,
+        rdma_task->allreduce_sliding_window.bufs->dst_ebuf,
+        gwbi_p->packed_dst_memh);
+    if (status != UCC_OK) {
+        tl_error(UCC_TASK_LIB(rdma_task), "failed to register dst memh: %s",
+                    ucc_status_string(status));
+        goto out;
```

Comment:
don't we need to unregister the src buffer here?

Reply:
If it fails here, then the schedule core will eventually call `ucc_tl_ucp_allreduce_sliding_window_rdma_task_finalize` which will unregister these